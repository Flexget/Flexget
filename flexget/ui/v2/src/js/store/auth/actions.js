const PREFIX = '@flexget/auth/';
export const LOGIN = `${PREFIX}LOGIN`;
export const LOGOUT = `${PREFIX}LOGOUT`;
