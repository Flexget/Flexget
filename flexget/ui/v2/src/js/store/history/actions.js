const PREFIX = '@flexget/history/';
export const GET_HISTORY = `${PREFIX}GET_HISTORY`;
