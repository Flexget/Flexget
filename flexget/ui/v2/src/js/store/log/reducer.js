import {
  LOG_CONNECT,
  LOG_MESSAGE,
  LOG_DISCONNECT,
  LOG_CLEAR,
} from 'store/log/actions';

const initState = {
  messages: [],
  connected: false,
};

export default function reducer(state = initState, action) {
  switch (action.type) {
    case LOG_CONNECT:
      return {
        ...state,
        messages: [],
        connected: true,
      };
    case LOG_MESSAGE:
      return {
        ...state,
        messages: [action.payload, ...state.messages],
      };
    case LOG_DISCONNECT:
      return {
        ...state,
        connected: false,
      };
    case LOG_CLEAR:
      return {
        ...state,
        messages: [],
      };
    default:
      return state;
  }
}
