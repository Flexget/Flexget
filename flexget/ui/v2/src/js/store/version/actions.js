const PREFIX = '@flexget/version/';
export const GET_VERSION = `${PREFIX}GET_VERSION`;
