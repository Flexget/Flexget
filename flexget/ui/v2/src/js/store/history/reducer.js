import { GET_HISTORY } from 'store/history/actions';

const initState = {
  totalCount: 0,
  items: [],
};


export default function reducer(state = initState, { type, payload }) {
  switch (type) {
    case GET_HISTORY:
      return {
        totalCount: payload.headers.get('total-count'),
        items: payload.refresh ? payload.data : [...state.items, ...payload.data],
      };
    default:
      return state;
  }
}
