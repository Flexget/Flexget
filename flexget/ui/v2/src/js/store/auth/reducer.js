import { LOGIN, LOGOUT } from 'store/auth/actions';
import { ERROR_STATUS } from 'store/status/actions';
import { GET_VERSION } from 'store/version/actions';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case LOGIN:
    case GET_VERSION:
      return {
        loggedIn: true,
      };
    case LOGOUT:
      return {};
    case ERROR_STATUS:
      if (action.payload.statusCode === 401) {
        return {};
      }
    default: // eslint-disable-line no-fallthrough
      return state;
  }
}
