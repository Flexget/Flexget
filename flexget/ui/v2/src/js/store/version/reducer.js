import { GET_VERSION } from 'store/version/actions';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case GET_VERSION: {
      const {
        api_version: api,
        flexget_version: flexget,
        latest_version: latest,
      } = action.payload;

      return { api, flexget, latest };
    }
    default:
      return state;
  }
}
