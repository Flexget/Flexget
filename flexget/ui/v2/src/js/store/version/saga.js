import { call, put, takeLatest } from 'redux-saga/effects';
import { delay } from 'redux-saga';
import { action, requesting } from 'utils/actions';
import { get } from 'utils/fetch';
import { GET_VERSION } from 'store/version/actions';

export function* getVersion() {
  try {
    yield call(delay, 500);
    const { data } = yield call(get, '/server/version');
    yield put(action(GET_VERSION, data));
  } catch (err) {
    yield put(action(GET_VERSION, err));
  }
}

export default function* saga() {
  yield takeLatest(requesting(GET_VERSION), getVersion);
}
