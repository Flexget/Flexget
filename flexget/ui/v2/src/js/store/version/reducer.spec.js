import reducer from 'store/version/reducer';
import { GET_VERSION } from 'store/version/actions';

describe('store/version/reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should get version on GET_VERSION', () => {
    expect(reducer(undefined, {
      type: GET_VERSION,
      payload: {
        api_version: '1.1.2',
        flexget_version: '2.10.11',
        latest_version: '2.10.60',
      },
    })).toMatchSnapshot();
  });
});
