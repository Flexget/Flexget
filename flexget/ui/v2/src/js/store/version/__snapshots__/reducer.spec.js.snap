// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/version/reducer should get version on GET_VERSION 1`] = `
Object {
  "api": "1.1.2",
  "flexget": "2.10.11",
  "latest": "2.10.60",
}
`;

exports[`store/version/reducer should return the initial state 1`] = `Object {}`;
